azure/table.ts

```diff
@@ -39,15 +46,237 @@ export class Table extends pulumi.ComponentResource implements cloud.Table {
             primaryKeyType = "string";
         }
 
+        if (primaryKeyType !== "string") {
+            throw new RunError("Only [string] is supported for [primaryKeyType] for an Azure [cloud.Table].");
+        }
+
         super("cloud:table:Table", name, {
             primaryKey: primaryKey,
             primaryKeyType: primaryKeyType,
         }, opts);
 
-        this.get = _ => { throw new Error("Method not implemented."); };
-        this.insert = _ => { throw new Error("Method not implemented."); };
-        this.scan = () => { throw new Error("Method not implemented."); };
-        this.delete = _ => { throw new Error("Method not implemented."); };
-        this.update = _ => { throw new Error("Method not implemented."); };
+        const primaryKeyOutput = pulumi.output(primaryKey);
+
+        const storageAccount = shared.getGlobalStorageAccount();
+
+        // Azure storage tables need a 'partition' to place data into.  Because we have no need to
+        // expose such functionality through the cloud.Table API, we just create a simple 'part1'
+        // partition and place all data there.
+        const partitionKey = "part1";
+
+        // The underlying azure table that will store all the data.
+        this.table = new azure.storage.Table(name, {
+            resourceGroupName: shared.globalResourceGroupName,
+            storageAccountName: storageAccount.name,
+        }, { parent: this});
+
+        const tableName = this.table.name;
+
+        this.get = async (query: any) => {
+            const rowKey = query[primaryKeyOutput.get()];
+            if (!rowKey) {
+                throw new RunError(`[query] must have a value specified for [${primaryKeyOutput.get()}]`);
+            }
+
+            const azureStorageSDK = await import("azure-storage");
```

Comment:
No.   I just personally like it when callbacks declare their own imports.  I find it helps me better remember that this is runtime code vs deployment time code.  That said, since we don't really push this pattern on users, perhaps i should just use the normal module-capture pattern.